mne_bids/tsv_handler.py

```diff
@@ -0,0 +1,106 @@
+
+
+def from_tsv(fname, dtypes=None):
+    """Read a tsv file into an OrderedDict
+
+    Parameters
+    ----------
+    fname : str
+        Path to the file being loaded.
+    dtypes : list
+        List of types to case the values loaded as. This is specified column by
+        column.
+        Defaults to None. In this case all the data is loaded as strings."""
+    data = np.loadtxt(fname, dtype=str, delimiter='\t')
+    column_names = data[0, :]
+    info = data[1:, :]
+    data_dict = OrderedDict()
+    if isinstance(dtypes, (list, tuple)):
+        if len(dtypes) == info.shape[1]:
+            for i, name in enumerate(column_names):
+                data_dict[name] = info[:, i].astype(dtypes[i]).tolist()
+    else:
+        for i, name in enumerate(column_names):
+            data_dict[name] = info[:, i].tolist()
+    return data_dict
+
+
+def to_tsv(data, fname):
+    """ Write an OrderedDict into a tsv file """
+    rows = len(data[list(data.keys())[0]])
+    with open(fname, 'w') as f:
+        f.write('\t'.join(list(data.keys())))
+        f.write('\n')
+        for i in range(rows):
+            row_data = list(str(data[key][i]) for key in data)
+            f.write('\t'.join(row_data))
+            f.write('\n')
+
+
+def combine(data1, data2, drop_column=None):
+    """Add two OrderedDict's together with the option of dropping repeated data
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Original array of data.
+    other : numpy.ndarray
+        New array of data.
+    drop_column : str
+        Name of the column to check for duplicate values in.
+        Any duplicates found will be dropped from the original data array (ie.
+        most recent value are kept).
+
+    """
+    for key, value in data2.items():
+        data1[key].extend(value)
+    if drop_column in data1:
+        column_data = data1[drop_column]
+        values_count = Counter(column_data)
+        for key, value in values_count.items():
+            # find the locations of the first n-1 values and remove the rows
+            for _ in range(value - 1):
+                idx = data1[drop_column].index(key)
+                for column in data1.values():
+                    del column[idx]
+
+
+def drop(data, values, column):
+    """Remove specified values in a column. This occurs in-place.
+
+    Parameters
+    ----------
+    data : collections.OrderedDict
+        Data to drop values from.
+    values : list
+        List of values to drop. Any row containing this value in the specified
+        column will be dropped.
+    column : string
+        Name of the column to check for the existence of `value` in.
+
+    """
+    mask = np.in1d(data[column], values, invert=True)
+    for key in data.keys():
+        data[key] = np.array(data[key])[mask].tolist()
+
+
+def contains_row(data, row_data):
+    """Whether the specified row data exists in the OrderedDict """
+    mask = None
+    for i, column_data in enumerate(data.values()):
```

Comment:
normally we avoid variable names with single letter. It can cause confusion ...

Reply:
I thought `i` or `j` were pretty common symbol for things like this though?

Reply:
I thought the same until I started contributing to MNE :)

Reply:
haha, ok, I'll fix it up along with the... list of other suggestions :P